File: src/classes/api/req_res/get-gkill-info-response.ts
'use strict'

import { GkillAPIResponse } from '../gkill-api-response'

export class GetGkillInfoResponse extends GkillAPIResponse {

    user_id: string

    device: string

    user_is_admin: boolean

    cache_clear_count_limit: number

    global_ip: string

    private_ip: string

	version: string

	build_time: Date

	commit_hash: string

    constructor() {
        super()
        this.user_id = ""
        this.device = ""
        this.user_is_admin = false
        this.cache_clear_count_limit = 1001
        this.global_ip = ""
        this.private_ip = ""
        this.version = ""
        this.build_time = new Date(0)
        this.commit_hash = ""
    }

}

File: src/app/gkill/api/req_res/get_gkill_info_response.go
package req_res

import (
	"time"

	"github.com/mt3hr/gkill/src/app/gkill/api/message"
)

type GetGkillInfoResponse struct {
	Messages []*message.GkillMessage `json:"messages"`

	Errors []*message.GkillError `json:"errors"`

	UserID string `json:"user_id"`

	Device string `json:"device"`

	UserIsAdmin bool `json:"user_is_admin"`

	CacheClearCountLimit int64 `json:"cache_clear_count_limit"`

	GlobalIP string `json:"global_ip"`

	PrivateIP string `json:"private_ip"`

	Version string `json:"version"`

	BuildTime time.Time `json:"build_time"`

	CommitHash string `json:"commit_hash"`
}
